### DISCUSSION TITLE
PayPal/Stripe: Track actual paid payment fees

### DISCUSSION THREAD

**Discussion by MacLake** [pretix/pretix#1964] [Ideas]
Currently you can only enter the fees for Paypal or other payment methods as an additional charge in the event configuration, but there are some problems with that:

- You can’t really calculate the Paypal fee because it depends on the country of the customer’s and merchant’s Paypal account. Within the Euro zone it’s 2.9 % + 0.35 € to a German account, but from other countries it’s significantly more. 

- The fee you enter is an additional charge shown to the customer. You may want that or not. We have e. g. a small event for which we will just offer payments with Paypal and we want to the fee to be included in the shown price.

- It would be very nice to have all payment data in Pretix including the real fees by Paypal or Stripe. Then you could do the bookkeeping just with these data. Without that information a colleague has to check the Paypal statements and do all these calculations in a spreadsheet. 

I don’t know much about the Paypal REST API, but with the one of the APIs used by [django-paypal ](https://github.com/spookylukey/django-paypal) you get [IPNs](https://django-paypal.readthedocs.io/en/stable/standard/ipn.html) including the real fee charged by Paypal denoted as mc_fee. Can we get this information also in the REST API? It would be great to have it in the database then.

**Comment by raphaelm** [comment]
Hi,

thank you very much for the suggestion! 

First of all, I want to stress a difference, just to make it 100% clear for other people who read this (I think I get what you are saying but others might not): The fee that can be currently configured in pretix is always **added** to the customers total (and therefore also shown. This is by design and is unlikely to change. For this purpose, yes, it is problematic that e.g. Stripe charges different amounts for European cards than for non-European. However, I don't see a way for either PayPal or Stripe to get the fees for the specific user *before* the transaction is performed. On the other hand, it also allows you to impose fees to discourage or encourage specific payment methods for non-monetary reasons. If you don't want to burden your customers with the payment fees, fine, just leave the fields empty.

What you are talking about is tracking the **actual** payment fees paid for **reporting**, regardless of if they are included in the ticket prices or not. Both the PayPal REST API and Stripe's API expose that data (I didn't check, but my memory says so). I agree that this would be interesting for accounting purposes.

Implementing this would include:
* Adding a field to the ``Order`` model to track this (easy)
* Extend the Stripe and PayPal implementations to automatically fill this field (probably easy)
* Build meaningful reports, integrate it into the order overview table in a non-confusing way (that's probably non-trivial)
* Decide (or make it configurable) to show sums (like the revenue statistics) with or without the fee (tricky from an UX- and from an "did I think of everything"-point of view).

Cheers
Raphael

**Comment by MacLake** [comment]
Hi Raphael,

thanks for your fast answer. Yes, I was talking about tracking the real fees charged by the providers after the payments. I don’t expect Pretix to get the exact fees in advance (I would expect from Paypal or Stripe that they have predictable fees, but that’s another thing). The configuration options for charging the user for different payment methods are just perfect, no need to change this.

My wish is just to track the fees actually charged by Paypal or Stripe in order not to check the statements for each booking, especially becaue we have participants from all over the world, i. e. different Papal fees.

Just to tell you my user experience when testing Pretix with Paypal sandbox accounts: I was bit confused since it was not clear for me whether the payment method fees in the CSVs were the ones charged by Paypal or the ones the customers were charged by Pretix. You’re right, having both in the reports in a non-confusing way is not easy, but just having also the real fees charged by Paypal and Stripe (if possible also possible for the latter) in the CSV export would already help a lot.

Anyway, thanks for the nice program!

Cheers,
Jens
